core/build/purge-config.js

```diff
@@ -4,7 +4,7 @@ const omit = require('lodash/omit')
  * clear config properties that shouldn't be visible on frontend
  */
 module.exports = function loader(source) {
-  let config = JSON.parse(source);
+  let config = typeof source === 'string' ? JSON.parse(source) : source;
```

Comment:
Can you create a new folder `purge-config` and there add files `purgeConfig.ts` and `purgeConfigLoader.ts`? Or something similar. Just to split purge logic and loader. In first one you can add everything that is between `let config = JSON.parse(source);` and `return JSON.stringify(config);`. In second you can add parse/stringify which related to webpack. First file can be then used directly in `core/server-entry.ts`